### PR TITLE
extend_linter_integration.md : Add more examples

### PR DESCRIPTION
**Pull request by Shade5**
Added a few more suggestions to make the project more
precise.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: data/projects/extend_linter_integration.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ wrapped together with concrete improvement proposals to the `@linter`
 decorator if needed. Because implementing a linter bear is a rather trivial
 task it is highly recommended to spice up the project with something
 interesting, like adding project wide linter support to the `@linter`
-decorator. Improving by adding more options can also be part of this.
+decorator or improving the testing API of linter bears or maybe even improving
```
Comment (meetmangukiya) [inline data/projects/extend_linter_integration.md]: I don't think we'd want a maybe here :)

**Comment by sims1253** [comment]
can you remove the first paragraph? The second one is a nice intro to what this is about.
Could you then add some motivation to the second one? In the mists of "This project would widen the range of existing linters coala supports and this make them available for usage in coala. This helps devs that had their favourite tools not supported before." Some motivation as in "what problem do we solve with this" and "what are the benefits from this"

also can you reorganize the milestones to fit the scheme you can see here? http://projects.coala.io/#/projects?project=integrate_antlr_v4_into_coala

gsoc has 3 coding parts now.

File: data/projects/extend_linter_integration.md [hunk 2]
```diff
@@ -1,39 +1,48 @@
...
-
 A very easy way to cover a broad spectrum of analysis routines
 is wrapping linters into coala. Currently we have the `@linter` decorator
 which simplifies this integration. Based on this decorator, a large selection
-of linters can be implemented rapidly. A tutorial on writing linter bears is
+of linters can be implemented rapidly. 
+
+This project would widen the range of existing linters coala supports and makes 
+them available for usage in coala. This helps devs that had their favourite tools
+not supported before. I also enhances the testing API of linters bears, which are
```
Comment (aalind0) [inline data/projects/extend_linter_integration.md]: Typo.   I -> It

Comment (sims1253) [inline data/projects/extend_linter_integration.md]: linters bears -> linter bears

File: data/projects/extend_linter_integration.md [hunk 3]
```diff
@@ -1,39 +1,48 @@
...
 wrapped together with concrete improvement proposals to the `@linter`
 decorator if needed. Because implementing a linter bear is a rather trivial
 task it is highly recommended to spice up the project with something
 interesting, like adding project wide linter support to the `@linter`
-decorator. Improving by adding more options can also be part of this.
+decorator or improving the testing API of linter bears or even improving
+the linter bear docs! Improving by adding more options can also be part of this.
 
 Upstream contributions to linters are encouraged, e.g. to add auto fixing
 abilities.
```
Comment (sims1253) [inline data/projects/extend_linter_integration.md]: abilities -> capabilities

File: data/projects/extend_linter_integration.md [hunk 4]
```diff
@@ -1,39 +1,48 @@
...
+##### PREPARATION/BONDING
 
 * The applicant has tried out all linters manually.
 * The applicant has outlined all changes proposed to the linter
   framework, if needed as a [cEP](https://coala.io/cep).
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
 * The first batch of linters have been merged.
```
Comment (sims1253) [inline data/projects/extend_linter_integration.md]: Make this "one linter bear is merged" so it's testable and describes the desired state

Comment (sims1253) [inline data/projects/extend_linter_integration.md]: "at least one linter bear" :D

File: data/projects/extend_linter_integration.md [hunk 5]
```diff
@@ -1,39 +1,48 @@
...
 * At least one improvement to the linter framework is in review state.
 * All of them are properly tested and documented.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
 * The second batch of linters have been merged.
```
Comment (sims1253) [inline data/projects/extend_linter_integration.md]: "At least two"? similar to above

File: data/projects/extend_linter_integration.md [hunk 6]
```diff
@@ -1,39 +1,48 @@
...
 * At least one improvement to the linter framework has been merged,
   if possible using newly implemented features to `@linter`.
 * All of them are properly tested and documented.
+
+##### CODING PHASE 3
+
+* The third batch of linters have been merged.
```
Comment (sims1253) [inline data/projects/extend_linter_integration.md]: "At least three" similar to above

File: data/projects/extend_linter_integration.md [hunk 7]
```diff
@@ -1,39 +1,48 @@
...
-
 A very easy way to cover a broad spectrum of analysis routines
 is wrapping linters into coala. Currently we have the `@linter` decorator
 which simplifies this integration. Based on this decorator, a large selection
-of linters can be implemented rapidly. A tutorial on writing linter bears is
+of linters can be implemented rapidly. 
+
+This project would widen the range of existing linters coala supports and makes 
+them available for usage in coala. This helps devs that had their favourite tools
+not supported before. It also enhances the testing API of linter bears, which are
```
Comment (SanketDG) [inline data/projects/extend_linter_integration.md]: Can we have "This helps devs whose favourite tools were not supported"?

**Comment by sims1253** [comment]
cherry-picked into https://github.com/coala/projects/pull/209

**Pull request closed without merging**